src/Microsoft.AspNet.Mvc.Razor.Host/MvcRazorParser.cs

```diff
@@ -107,9 +125,43 @@ private class GlobalImportAddRemoveTagHelperVisitor : TagHelperDirectiveSpanVisi
                 IEnumerable<TagHelperDirectiveDescriptor> descriptors,
                 ParserErrorSink errorSink)
             {
-                return base.GetTagHelperDescriptorResolutionContext(
-                    _globalImportDirectiveDescriptors.Concat(descriptors),
-                    errorSink);
+                var directivesToImport = MergeDirectiveDescriptors(descriptors, _globalImportDirectiveDescriptors);
+
+                return base.GetTagHelperDescriptorResolutionContext(directivesToImport, errorSink);
+            }
+
+            private static IEnumerable<TagHelperDirectiveDescriptor> MergeDirectiveDescriptors(
+                IEnumerable<TagHelperDirectiveDescriptor> descriptors,
+                IEnumerable<TagHelperDirectiveDescriptor> inheritedDescriptors)
+            {
+                var mergedDescriptors = new List<TagHelperDirectiveDescriptor>();
+                TagHelperDirectiveDescriptor prefixDirectiveDescriptor = null;
+
+                foreach (var descriptor in inheritedDescriptors)
+                {
+                    if (descriptor.DirectiveType != TagHelperDirectiveType.TagHelperPrefix)
+                    {
+                        mergedDescriptors.Add(descriptor);
+                    }
+                    else
+                    {
+                        // Always take the latest @tagHelperPrefix descriptor. Can only have 1 per page.
+                        prefixDirectiveDescriptor = descriptor;
```

Comment:
It will fail to parse the `_GlobalImport` if it happens to have more than one. As for multiple existing in the hierarchy this is supported because they override each other.